File: tlb/05/solutions/simple_head0.py
#!/usr/bin/python

import re, sys

n_lines = 10

if len(sys.argv) > 1 and re.match(r'-[0-9]+', sys.argv[1]):
    arg = sys.argv[1]
    arg = arg[1:] # remove first character
    n_lines = int(arg)

n = 1
for line in sys.stdin:
    if n > n_lines:
        break
    sys.stdout.write(line)
    n += 1

File: tlb/05/solutions/simple_head0.pl
#!/usr/bin/perl -w
$n_lines = 10;
if (@ARGV && $ARGV[0] =~ /-[0-9]+/) {
    $n_lines = $ARGV[0];
    $n_lines =~ s/-//;
    shift @ARGV;
}
$n = 1;
while (<STDIN>) {
    if ($n++ > $n_lines) {
        last;
    }
    print;
}
